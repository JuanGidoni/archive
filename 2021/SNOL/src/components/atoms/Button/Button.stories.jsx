import React from "react";
import Button from "./index";

export default {
  title: "Button",
  argTypes: {
    className: {
      options: ['flex-1 bg-blue-800 text-white p-2 rounded ml-3', 'bg-gray-300 p-2 rounded shadow-sm'],
      control: { type: 'radio' }
    },
  },
  args: {
    text: "Send"
  },
  component: Button,
};


export const ButtonExample = (args) => (
  <Button {...args} />
);

