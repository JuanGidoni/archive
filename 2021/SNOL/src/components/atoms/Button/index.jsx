const Button = ({ className, type, text, handleEvent }) => {
  return (
    <button type={type} className={className} onClick={handleEvent}>
      {text}
    </button>
  );
};

export default Button;
