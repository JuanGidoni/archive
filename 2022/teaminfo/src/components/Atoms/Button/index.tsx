import { ButtonProps } from "../../Types";

const Button = ({ children, onClick, className, type }: ButtonProps) => {
  return (
    <button
      type={type || "button"}
      className={className}
      onClick={() => onClick}
    >
      {children}
    </button>
  );
};

export default Button;
