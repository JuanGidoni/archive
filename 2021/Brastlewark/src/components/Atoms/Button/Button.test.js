import React from 'react';
import renderer from 'react-test-renderer';
import Button from './Button';

const onClick = jest.fn();

describe('<Button />', () => {
 it('renders correctly and match snapshot', () => {
  const tree = renderer
   .create(
    <Button type="button" className="btn btn-primary" onClick={onClick}>
     Example
    </Button>
   ).toJSON();
  expect(tree).toMatchSnapshot();
 });
});