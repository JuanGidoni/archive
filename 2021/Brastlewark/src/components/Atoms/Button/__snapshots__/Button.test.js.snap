// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly and match snapshot 1`] = `
<button
  className="btn btn-primary"
  onClick={[MockFunction]}
  type="button"
>
  Example
</button>
`;
